EIPS/eip-5299.md

```diff
@@ -0,0 +1,168 @@
+---
+eip: 5299
+title: Layer3 scaling pattern
+description: A primitive to facilitate eternal upgradeability of smart contracts, by making storage slots dynamic.
+author: Vinay Kumar (@v9hstk), Pratik Agarwal (@Prtk418)
+discussions-to: https://ethereum-magicians.org/t/eip-9000-layer3-scaling-standard/9039
+status: Draft
+type: Standards Track
+category: ERC
```

Comment:
```suggestion
type: Informational
```

Reply:
I'm not a fan of turning `Informational` EIPs into a "collection of Best Practices and Good Ideas".  At best, we'll end up with a mix of some good ideas, and some bad ideas since we don't have any means of curation (nor do I think editors should get into the game of Good Idea curation).

Reply:
> ```suggestion
> 
> type: Informational
> 
> ```

Please refer to the implementation examples provided in the assets. 

The key value add this design pattern enables is to induce composability in the storage structure, a key missing component in all the upgradable patterns. 

I will add additional implementation references describing storage scalability in low level upgradable patterns.

Reply:
@MicahZoltu Thanks for your comments on our Ethereum Improvement Proposal. Everyone is entitled to their opinion irrespective of its relevance, or wisdom.

If you have specific feedback on the proposed design short-comings, such as probable oversights, or edge cases that the developers might have overseen, me & my team of developers are happy to look into it.

If you have trouble understanding the scale & scope of value-add such a simplistic looking design pattern could bring to the ecosystem, you can look at Uniswap v1,v2,v3 as an example of the world that's deprived of this EIP. All upgradable patterns today enable functional and/or logic upgradability. By solving for storage scalability, we are able to make upgradable patterns lot more composable and enable a means to truly scale smart contracts.

 If you still have trouble, I'm happy to connect with you over a call, or probably a twitter space even, to dissect the proposed EIP in greater detail.